kernel/comps/virtio/src/device/block/device.rs

```diff
@@ -186,31 +185,40 @@ impl DeviceInner {
             device_id: None,
         };
 
+        fn config_space_change(_: &TrapFrame) {
+            info!("Virtio block device config space change");
+        }
+        fn dummy_handle_block_device(_: &TrapFrame) {
+            info!("Virtio block device handle block device");
+        }
+        device
+            .transport
+            .register_cfg_callback(Box::new(config_space_change))
+            .unwrap();
+        device
+            .transport
+            .register_queue_callback(0, Box::new(dummy_handle_block_device), false)
+            .unwrap();
+
+        // Sets the queue callback function properly before calling `get_id`;
+        // otherwise, the interrupt mechanism cannot work.
         let device_id = device.get_id();
```

Comment:
The verb get is usually reserved for light-weight methods. This `get_id` method involves I/O and blocking. So it should be considered heavy-weight. I suggest renaming it to `fetch_id` or `request_device_id`.

Reply:
Done